packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -37,7 +36,17 @@ type HTTPSubscriptionLinkOptions<TRoot extends AnyClientTypes> = {
   /**
    * EventSource options or a callback that returns them
    */
-  eventSourceOptions?: CallbackOrValue<EventSourceInit>;
+  eventSourceOptions?: () => EventSourceInit | Promise<EventSourceInit>;
+  /**
+   * For a given error, should we reinitialize the underlying EventSource?
+   *
+   * This is useful where a long running subscription might be interrupted by a recoverable network error,
+   * but the existing authorization header might have expired in the mean-time
+   */
+  shouldReinitialize?: (
+    status: number,
+    error: Event,
+  ) => boolean | Promise<boolean>;
```

Comment:
What if there's a transient error in the `eventSourceOptions`-callback? Should that follow the same code path?

Reply:
I'd see that as a userland problem and we just let it bubble